addons/web_editor/controllers/main.py

```diff
@@ -688,20 +730,15 @@ def save_library_media(self, media):
 
         return attachments
 
-    @http.route("/web_editor/get_ice_servers", type='json', auth="user")
-    def get_ice_servers(self):
-        return request.env['mail.ice.server']._get_ice_servers()
+    @http.route("/web_editor/get_editor_collaboration_infos", type='json', auth="user")
+    def get_editor_collaboration_infos(self, model_name, field_name, res_id):
+        signatures = generate_channel_signatures(request.env, model_name, field_name, res_id)
+        return {
+            'ice_servers': request.env['mail.ice.server'].sudo()._get_ice_servers(),
+            'next_expiration_timestamp': signatures['next_expiration_timestamp'],
+            'signatures': signatures['signatures'],
+        }
 
     @http.route("/web_editor/bus_broadcast", type="json", auth="user")
-    def bus_broadcast(self, model_name, field_name, res_id, bus_data):
-        document = request.env[model_name].browse([res_id])
-
-        document.check_access_rights('read')
-        document.check_field_access_rights('read', [field_name])
-        document.check_access_rule('read')
-        document.check_access_rights('write')
-        document.check_field_access_rights('write', [field_name])
-        document.check_access_rule('write')
-
-        channel = (request.db, 'editor_collaboration', model_name, field_name, int(res_id))
-        request.env['bus.bus'].sendone(channel, bus_data)
+    def bus_broadcast(self, signature, bus_data):
+        request.env['bus.bus'].sendone(f"editor_collaboration_{signature}", bus_data)
```

Comment:
Is this still necessary? Can't the client just do that directly?

Reply:
yes I prefer to have an explicit route for that. However it deserves a better name.
How about something like this?

```patch
- @http.route("/web_editor/bus_broadcast", type="json", auth="user")
+ @http.route("/web_editor/edit/notify", type="json", auth="user")
```